lms/djangoapps/discussion/rest_api/views.py

```diff
@@ -578,6 +583,79 @@ def partial_update(self, request, comment_id):
         return Response(update_comment(request, comment_id, request.data))
 
 
+class IsEnrolledOrIsStaff(permissions.BasePermission):
+    """Permission that checks to see if the user is enrolled in the course or is staff."""
+
+    def has_permission(self, request, view):
+        """Returns true if the user is enrolled or is staff."""
+        course_key = CourseKey.from_string(view.kwargs.get('course_id'))
+        return has_team_api_access(request.user, course_key)
```

Comment:
If you think this access method is helpful here, I'd encourage you to refactor it so that it can be used in teams app as well discussions API. Currently its not making sense that we are calling `has_team_api_access` in discussions api. what do you think? 

Also, this endpoint is not allowing course teams to upload any content. What roles you would want to allow to hit this endpoint for uploading file?

I think same set of users who can access discussions / post discussions / etc.

Reply:
I used the same function because it's checking the same roles that would make sense here, but you're right that it's not communicating the correct idea.
I made it check the relevant roles explicitly, which makes it cleaner.

I'm currently checking for global staff, course staff, course instructors, enrolled users, and users with discussions privileges (as checked by `has_discussion_privileges`). What other roles should be allowed?